File: android/src/main/java/expo/modules/mediapipepose/PoseCameraOptions.kt
package expo.modules.mediapipepose

data class PoseCameraOptions(
    val facing: String = "front",
    val lens: String = "auto",
    val zoom: Double = 1.0,
    val previewFps: Int = 30,
    val modelVariant: String = "full",
    val modelPath: String? = null,
    val rotation: Int = 0,
    val viewWidth: Int = 0,
    val viewHeight: Int = 0,
    val minPoseDetectionConfidence: Double = 0.35,
    val minPosePresenceConfidence: Double = 0.35,
    val minTrackingConfidence: Double = 0.35,
) {
    fun isValid(): Boolean {
        if (facing !in listOf("front", "back")) return false
        if (lens !in listOf("auto", "wide")) return false
        if (modelVariant !in listOf("lite", "full", "heavy")) return false
        if (!zoom.isFinite() || zoom < 1) return false
        if (previewFps !in 1..60) return false
        val confidenceValues =
            listOf(minPoseDetectionConfidence, minPosePresenceConfidence, minTrackingConfidence)
        return confidenceValues.all { it.isFinite() && it in 0.0..1.0 }
    }
}

File: ios/PoseCameraOptions.swift
import AVFoundation

struct PoseCameraOptions: Equatable {
  var facing = "front"
  var lens = "auto"
  var zoom = 1.0
  var frameLimit = 30
  var modelVariant = "full"
  var modelPath: String?
  var orientation = AVCaptureVideoOrientation.portrait
  var minPoseDetectionConfidence = 0.35
  var minPosePresenceConfidence = 0.35
  var minTrackingConfidence = 0.35
  var isValid: Bool {
    guard ["front", "back"].contains(facing) else { return false }
    guard ["auto", "wide", "ultraWide"].contains(lens) else { return false }
    guard ["lite", "full", "heavy"].contains(modelVariant) else { return false }
    guard zoom.isFinite && zoom >= 1 else { return false }
    guard (1...60).contains(frameLimit) else { return false }
    let confidenceValues = [
      minPoseDetectionConfidence, minPosePresenceConfidence, minTrackingConfidence,
    ]
    return confidenceValues.allSatisfy { $0.isFinite && (0...1).contains($0) }
  }
}
